src/main/java/ch/njol/skript/expressions/ExprEventExpression.java

```diff
@@ -39,22 +42,29 @@
  */
 @NoDoc
 public class ExprEventExpression extends WrapperExpression<Object> {
+
 	static {
-		Skript.registerExpression(ExprEventExpression.class, Object.class, ExpressionType.PROPERTY, "[the] event-%*classinfo%");// property so that it is parsed after most other expressions
+		Skript.registerExpression(ExprEventExpression.class, Object.class, ExpressionType.PROPERTY, "[the] event-<.+>");// property so that it is parsed after most other expressions
```

Comment:
This doesn't need to be changed. A classinfo will define if it can be used as a plural or not. The classinfos should be updated accordingly, not this syntax.

Reply:
You wouldn't be able to tell whether they used a single class info or a plural one if you use the classinfo type instead, it'll just give you the classinfo.

Reply:
A method can be added to the ClassInfo that defines if it was used as a plural or not.

Reply:
That's not how it works, as getting a ClassInfo object is not instantiating a new object, it retrieves it from a list of registered ClassInfos. Unless we change how retrieving ClassInfos work, then we can't do that

Reply:
It can be added to the parse context that is used in the parse.

Reply:
What do you mean?